tutorial/markdown/python/python-haystack-pdf-chat/python-haystack-pdf-chat.md

```diff
@@ -0,0 +1,497 @@
+- **Prompt Building**: Haystack's [PromptBuilder](https://docs.haystack.deepset.ai/docs/promptbuilder) component allows you to create custom prompts that guide the language model's behavior and output.
+- **Streaming Output**: LangChain supports [streaming](https://python.langchain.com/docs/expression_language/streaming/), allowing the app to stream the generated answer to the client in real-time.
+
+By combining Vector Search with Couchbase, RAG, and Haystack, the PDF Chat app can efficiently ingest PDF documents, convert their content into searchable embeddings, retrieve relevant information based on user queries and conversation context, and generate context-aware and informative responses using large language models.
+
+## Let us Understand the Flow
+
+To begin this tutorial, clone the repo and open it up in the IDE of your choice. Now you can learn how to create the PDF Chat App. The whole code is written in `chat_with_pdf.py` file.
+
+### App Flow
+
+The fundamental workflow of the application is as follows: The user initiates the process from the Main Page's sidebar by uploading a PDF. This action triggers the `save_to_vector_store` function, which subsequently uploads the PDF into the Couchbase vector store. Following this, the user can now chat with the LLM.
+
+On the Chat Area, the user can pose questions. These inquiries are processed by the Chat API, which consults the LLM for responses, aided by the context provided by RAG. The assistant then delivers the answer, and the user has the option to ask additional questions.
+
+![App Flow](python_app_flow.png)
+
+## Connecting to Couchbase
+
+The first step is connecting to Couchbase. Couchbase Vector Search is required for PDF Upload as well as during chat (For Retrieval). We will use the Haystack CouchbaseDocumentStore to connect to the Couchbase cluster. The connection is established in the `get_document_store` function.
+
+The connection string and credentials are read from the environment variables. We perform some basic required checks for the environment variable not being set in the `secrets.toml`, and then proceed to connect to the Couchbase cluster. We connect to the cluster using [connect](https://docs.couchbase.com/python-sdk/current/hello-world/start-using-sdk.html#connect) method.
+
+```python
+@st.cache_resource(show_spinner="Connecting to Vector Store")
+def get_document_store():
+    """Return the Couchbase document store"""
+    return CouchbaseDocumentStore(
+        cluster_connection_string=Secret.from_env_var("DB_CONN_STR"),
+        authenticator=CouchbasePasswordAuthenticator(
+            username=Secret.from_env_var("DB_USERNAME"),
+            password=Secret.from_env_var("DB_PASSWORD")
+        ),
+        cluster_options=CouchbaseClusterOptions(profile='wan_development'),
+        bucket=os.getenv("DB_BUCKET"),
+        scope=os.getenv("DB_SCOPE"),
+        collection=os.getenv("DB_COLLECTION"),
+        vector_search_index=os.getenv("INDEX_NAME"),
+    )
+```
```

Comment:
You can also add a couple of imports which are being used in this snippet. Like import CouchbaseDocumentStore, CouchbasePasswordAuthenticator, CouchbaseClusterOptions.

Just makes it a bit easier to understand